### PR TITLE
Removed fittable; added new options

### PR DESCRIPTION
**Pull request by rafabernad**
I don't feel that a widget should make use of fittable (and had issues putting it as a toolbar component), so I've removed it.

Also, I've added an optional icon, and added options to hide caption and/or arrow.

### CODE REVIEW

**Comment by MaKleSoft** [comment]
Thanks for contributing! I definitely agree regarding the use of the fittable layout. The other additions also seem useful. However please try not to make any changes to the original code formatting. For example, there are a lot of injected whitespaces that add unnecessary line changes. As a rule, try to keep the number of changed lines to a minimum. Please also try to keep the original formatting of the components array. I've also added some comments in the diff section.

Finally, in the current implementation the control inherits the text-align property from its parent. Since the caption should generally be aligned left you might want to explicitly add 'text-align: left' to the select-decorator class. But I can also add this later.

File: v2/SelectDecorator/SelectDecorator.js [hunk 1]
```diff
@@ -1,35 +1,61 @@
 enyo.kind({
-	name: "SelectDecorator",
-	classes: "onyx-button select-decorator",
-	handlers: {
-		onchange: "changeHandler"
+	name : "SelectDecorator",
```
Comment (MaKleSoft) [inline v2/SelectDecorator/SelectDecorator.js]: Injected whitespaces break the formatting pattern and add unnecessary line changes.

File: v2/SelectDecorator/SelectDecorator.js [hunk 2]
```diff
@@ -1,35 +1,61 @@
...
 		var select = this.getClientControls()[0];
 
-		if (select) {
+		if(select) {
 			this.changeHandler(select);
 		}
 	},
-	changeHandler: function(sender, event) {
+	changeHandler : function(sender, event) {
+		//if(this.showCaption) {
```
Comment (MaKleSoft) [inline v2/SelectDecorator/SelectDecorator.js]: Avoid committing commented lines if possible

File: v2/SelectDecorator/SelectDecorator.js [hunk 3]
```diff
@@ -1,35 +1,61 @@
...
 		this.$.innerText.setContent(caption);
+		//}
 	},
-	components: [
-		{kind: "FittableColumns", noStretch: true, classes: "select-decorator-inner", components: [
-			{name: "innerText", fit: true, classes: "select-decorator-inner-text"},
-			{classes: "select-decorator-inner-arrow"}
-		]}
-	]
+	components : [{
```
Comment (MaKleSoft) [inline v2/SelectDecorator/SelectDecorator.js]: Please try to maintain the original formatting

**Comment by rafabernad** [comment]
Sorry for all the mess... Seems I reformatted the code on error... ;">

**Comment by MaKleSoft** [comment]
MERGED! Thanks again for contributing!

**Pull request merged**